tools/roslaunch/src/roslaunch/depends.py

```diff
@@ -94,20 +95,16 @@ def _get_arg_value(tag, context):
     else:
         raise RoslaunchDepsException("No value for arg [%s]"%(name))
 
-def _parse_arg(tag, context):
-    name = tag.attributes['name'].value
+def _check_ifunless(tag, context):
     if tag.attributes.has_key('if'):
         val = resolve_args(tag.attributes['if'].value, context)
-        if val == '1' or val == 'true':
-            return (name, _get_arg_value(tag, context))
-    elif tag.attributes.has_key('unless'):
+        if not convert_value(val, 'bool'):
+            return False
+    if tag.attributes.has_key('unless'):
```

Comment:
Previously if a tag had both - an `if` as well as an `unless` attribute the second one wasn't considered. Is there a reason to change this?